controller/api/models.py

```diff
@@ -353,6 +367,58 @@ def _start_containers(self, to_add):
         if set([c.state for c in to_add]) != set(['up']):
             err = 'warning, some containers failed to start'
             log_event(self, err, logging.WARNING)
+        # if the user specified a health check, try checking to see if it's running
+        try:
+            config = self.config_set.latest()
+            if 'HEALTHCHECK_URL' in config.values.keys():
+                self._healthcheck(to_add, config.values)
+        except Config.DoesNotExist:
+            pass
+
+    def _healthcheck(self, containers, config):
+        # if at first it fails, back off and try again at 10%, 50% and 100% of INITIAL_DELAY
+        intervals = [1.0, 0.1, 0.5, 1.0]
+        # HACK (bacongobbler): we need to wait until publisher has a chance to publish each
```

Comment:
Yep, sounds fair to me :)

On Mon, 3 Aug 2015 21:19 Matthew Fisher <notifications@github.com> wrote:

> In controller/api/models.py
> <https://github.com/deis/deis/pull/4127#discussion_r36120420>:
>
> > @@ -353,6 +367,58 @@ def _start_containers(self, to_add):
> >          if set([c.state for c in to_add]) != set(['up']):
> >              err = 'warning, some containers failed to start'
> >              log_event(self, err, logging.WARNING)
> > +        # if the user specified a health check, try checking to see if it's running
> > +        try:
> > +            config = self.config_set.latest()
> > +            if 'HEALTHCHECK_URL' in config.values.keys():
> > +                self._healthcheck(to_add, config.values)
> > +        except Config.DoesNotExist:
> > +            pass
> > +
> > +    def _healthcheck(self, containers, config):
> > +        # if at first it fails, back off and try again at 10%, 50% and 100% of INITIAL_DELAY
> > +        intervals = [1.0, 0.1, 0.5, 1.0]
> > +        # HACK (bacongobbler): we need to wait until publisher has a chance to publish each
>
> @Blystad <https://github.com/Blystad> for the sake of getting this PR
> through the door I'll keep the current implementation as-is, but I'll
> follow up in a future PR to use a polling solution. Does that sound fair to
> you?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/deis/deis/pull/4127/files#r36120420>.
>